### PR TITLE
DSWx-S1 SAS Final Integration

### LINKED ISSUE #449: [Bug]: DSWx-S1 ISO Metadata Discrepencies with Product Spec
### Checked for duplicates

Yes - I've already checked

### Describe the bug

As part of V&V testing for the DSWx-S1 PGE, the following discrepancies were reported between the ISO.xml and the metadata as described within the Product Spec Document (PSD):

1. ISO.xml specified an input of NOAA GSHHS for shoreline shape file, but nothing mentioned in the PSD
2. Could not find PROCESSING_INFORMATION_WORKFLOW in iso.xml 
3. PROCESSING_INFORMATION_THRESHOLD_TILE_AVERAGE in PSD is ProcessingInformationTileAverage in iso.xml (no Threshold) 
4. PROCESSING_INFORMATION_FILTER_WINDOW_SIZE in PSD is ProcessingInformationWindowSize (no Filter)

### What did you expect?

For finding 1, any references to the shoreline shapefile ancillary within the ISO.xml template for DSWx-S1 should be removed.

For finding 2, the metadata within a sample DSWx-S1 product needs to be examined to see if it is missing from there, or just from the iso.xml. If it is missing from the product, then ADT should update the Product Spec Document to remove PROCESSING_INFORMATION_WORKFLOW

For findings 3 and 4, the names used in the ISO.xml should be updated to match the names used in the PSD

### Reproducible steps

```bash
1.
2.
3.
...
```

### Environment

```bash
- Version of this software [e.g. vX.Y.Z]
- Operating System: [e.g. MacOSX with Docker Desktop vX.Y]
...
```

### LINKED ISSUE #461: [Bug]: InputGranules Metadata Field in DSWx-S1 Product Metadata has TMI
### Checked for duplicates

Yes - I've already checked

### Describe the bug

When requesting metadata for a given DSWx-S1 product from CMR, I get back values for the InputGranules field that contain system environment information, i.e. `/home/dswx_urs/input_dir/.../` that is irrelevant to most users. We should remove that information from the metadata we submit to CMR.

#### Send a query to CMR UAT for DSWx-S1 product metadata for a given tile
```
curl "https://cmr.uat.earthdata.nasa.gov/search/granules.umm_json?short_name=OPERA_L3_DSWX-S1_PROVISIONAL_V0&temporal[]=2024-05-12T08:10:00.000Z,2024-05-12T12:59:59.000Z&page_size=1000&provider=POCLOUD" | jq '.items[] | select(.umm.AdditionalAttributes[] | .Name == "MGRS_TILE_ID" and .Values[] == "T23WPQ")'
```

#### Result 

The below is a snippet for `InputGranules` field. Run the original query for the full result.

```
"InputGranules": [
      "/home/dswx_user/input_dir/T039-082293-IW1/OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0.h5",
      "/home/dswx_user/input_dir/T039-082293-IW1/OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0_HH.tif",
      "/home/dswx_user/input_dir/T039-082293-IW1/OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0_HV.tif",
      "/home/dswx_user/input_dir/T039-082293-IW1/OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0_mask.tif",
      "/home/dswx_user/input_dir/T039-082293-IW2/OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0.h5",
      "/home/dswx_user/input_dir/T039-082293-IW2/OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0_HH.tif",
      "/home/dswx_user/input_dir/T039-082293-IW2/OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0_HV.tif",
      "/home/dswx_user/input_dir/T039-082293-IW2/OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0_mask.tif",
```

### What did you expect?

I expect the following metadata for the `InputGranules` field:

```
"InputGranules": [
      "OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0.h5",
      "OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0_HH.tif",
      "OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0_HV.tif",
      "OPERA_L2_RTC-S1_T039-082293-IW1_20240512T090356Z_20240512T141542Z_S1A_30_v1.0_mask.tif",
      "OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0.h5",
      "OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0_HH.tif",
      "OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0_HV.tif",
      "OPERA_L2_RTC-S1_T039-082293-IW2_20240512T090357Z_20240512T141542Z_S1A_30_v1.0_mask.tif",
```

### Reproducible steps

See "Describe a bug" section for reproducing the metadata.

### Environment

My laptop.

### LINKED ISSUE #467: DSWx-S1 SAS Final PGE Integration
Ticket to track integration of the DSWx-S1 SAS Final delivery with the PGE library.

### PR DESCRIPTION
**Pull request by collinss-jpl**
## Description
- This branch updates the DSWx-S1 PGE code to accommodate the Final SAS delivery. Updates with this integration include validation of the new GLAD Classification ancillary map and an update to the ISO XML to bring it line with the product metadata available in the Final release.
- This branch also fixes several discrepancies between the product metadata (according to the product spec) and the ISO XML template that were identified during testing of the previous release
- Lastly, this branch fixes an issue in the Catalog Metadata generation, where paths relative to the PGE container were included with the lists of input and ancillary files

## Affected Issues
- Resolves #467
- Fixes #449 
- Fixes #461 

## Testing
- Unit test suite for DSWx-S1 has been updated to accommodate changes made for Final delivery
- Integration test for DSWx-S1 has been run on Jenkins with all product comparisons passing

### CODE REVIEW

File: .ci/scripts/dswx_s1/dswx_s1_final_1.0_runconfig.yaml [hunk 1]
```diff
@@ -167,9 +170,9 @@ RunConfig:
...
             set_not_water_to_nodata: False
-            set_hand_mask_to_nodata: False
-            set_layover_shadow_to_nodata: False
+            set_hand_mask_to_nodata: True
+            set_layover_shadow_to_nodata: True
             set_ocean_masked_to_nodata: False
             save_tif_to_output: True
 
-          log_file: /home/dswx_user/output_dir/dswx-s1.log
+          log_file:
```
Comment (JimHofman) [inline .ci/scripts/dswx_s1/dswx_s1_final_1.0_runconfig.yaml:178]: Just pointing out there is no path here.

File: src/opera/pge/base/base_pge.py [hunk 2]
```diff
@@ -300,8 +300,8 @@ def _create_catalog_metadata(self):
             'PGE_Name': self.runconfig.pge_name,
             'PGE_Version': self.PGE_VERSION,
             'SAS_Version': self.SAS_VERSION,
-            'Input_Files': self.runconfig.get_input_filenames(),
-            'Ancillary_Files': self.runconfig.get_ancillary_filenames(),
+            'Input_Files': list(map(basename, self.runconfig.get_input_filenames())),
+            'Ancillary_Files': list(map(basename, self.runconfig.get_ancillary_filenames())),
```
Comment (JimHofman) [inline src/opera/pge/base/base_pge.py:304]: Nice:  I assume this obviates the need to use 'basename' later.

File: src/opera/pge/dswx_s1/dswx_s1_pge.py [hunk 3]
```diff
@@ -253,6 +254,40 @@ def _validate_output(self):
 
             self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
 
+    def _checksum_output_products(self):
+        """
+        Generates a dictionary mapping output product file names to the
+        corresponding MD5 checksum digest of the file's contents.
+
+        The output products to generate checksums for is determined by scanning
```
Comment (JimHofman) [inline src/opera/pge/dswx_s1/dswx_s1_pge.py:262]: Suggested re-wording:  Determines the output products from which to generate checksums  by scanning the output product location ....

**Review by JimHofman — COMMENTED** [review]
Looks great Scott.  Just a couple of trivial comments.

File: .ci/scripts/dswx_s1/dswx_s1_final_1.0_runconfig.yaml (hunk 1 above)
Comment (collinss-jpl) [reply .ci/scripts/dswx_s1/dswx_s1_final_1.0_runconfig.yaml:178]: This was intentional, as we do not want this log file to be validated as an output product

**Review by RKuttruff — APPROVED** [review]
Looks good!

File: src/opera/pge/base/base_pge.py (hunk 2 above)
Comment (riverma) [reply src/opera/pge/base/base_pge.py:304]: Nice fix 👍

**Pull request merged**